pageserver/src/tenant/storage_layer/image_layer.rs

```diff
@@ -855,3 +914,133 @@ impl Drop for ImageLayerWriter {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use bytes::Bytes;
+    use pageserver_api::{
+        key::Key,
+        shard::{ShardCount, ShardIdentity, ShardNumber, ShardStripeSize},
+    };
+    use utils::{id::TimelineId, lsn::Lsn};
+
+    use crate::{tenant::harness::TenantHarness, DEFAULT_PG_VERSION};
+
+    use super::ImageLayerWriter;
+
+    #[tokio::test]
+    async fn test_rewrite() -> anyhow::Result<()> {
```

Comment:
I do not understand why use `-> anyhow::Result<()>` when it's mixed together with `unwrap`s and `assert!`. There is no upside to using `?` as I understand, it's always a less precise location than a panic would have.

Reply:
Changed in 5db0eddf5